pkg/network/ebpf/c/ip.h

```diff
@@ -112,7 +112,19 @@ __maybe_unused static __always_inline __u64 read_conn_tuple_skb(struct __sk_buff
     return 1;
 }
 
-// On older kernels, clang can generate Wunused-function warnings on static inline functions defined in 
+__maybe_unused static __always_inline bool is_equal(conn_tuple_t *t, conn_tuple_t *t2) {
+    return t->saddr_l == t2->saddr_l &&
+        t->saddr_h == t2->saddr_h &&
+        t->daddr_l == t2->daddr_l &&
+        t->daddr_h == t2->daddr_h &&
+        t->sport == t2->sport &&
+        t->dport == t2->dport &&
+        t->netns == t2->netns &&
+        t->pid == t2->pid &&
+        t->metadata == t2->metadata;
```

Comment:
looks like you're comparing the whole struct so can't we use something like `bpf_memcmp`?

Reply:
done